.github/workflows/deploy.yml

```diff
@@ -0,0 +1,121 @@
+name: deploy
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - lab
+      - master
+      - published
+
+# these permissions are needed to interact with GitHub's OIDC Token endpoint.
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  publish:
+    runs-on: ubuntu-20.04
+    steps:
+      -
+        name: Prepare
+        run: |
+          JEKYLL_ENV=development
+          DOCS_AWS_REGION=us-east-1
+          if [ "${{ github.ref }}" = "refs/heads/master" ]; then
+            DOCS_URL="https://docs-stage2.docker.com"
+            DOCS_AWS_IAM_ROLE=""
+            DOCS_S3_BUCKET=""
+            DOCS_S3_CONFIG="s3-config.json"
+            DOCS_CLOUDFRONT_ID=""
+            DOCS_LAMBDA_FUNCTION_REDIRECTS=""
+            DOCS_SLACK_MSG="Successfully deployed docs-stage2 from master branch. https://docs-stage2.docker.com/"
```

Comment:
@VictorBersy Let me know when you have the IAM role, bucket, cloudfront ID and lambda name to use for stage env. I already set `DOCS_URL="https://docs-stage2.docker.com"` as discussed.

Reply:
Here's the requested information:

```
DOCS_AWS_IAM_ROLE="arn:aws:iam::710015040892:role/stage-docs-docker.github.io-20220816140248629900000003"
DOCS_S3_BUCKET="stage-docs-docker.github.io"
DOCS_CLOUDFRONT_ID="E1R7CSW3F0X4H8"
DOCS_LAMBDA_FUNCTION_REDIRECTS="DockerDocsRedirectFunction-stage-role"
```

Reply:
Thanks! Added stage aws config in last commit. Let me know when you have prod ready and I update the config for it too.

Reply:
And for prod:
```
DOCS_AWS_IAM_ROLE="arn:aws:iam::710015040892:role/prod-docs-docker.github.io-20220816161549883800000001"
DOCS_S3_BUCKET="prod-docs-docker.github.io"
DOCS_CLOUDFRONT_ID="E228TTN20HNU8F"
DOCS_LAMBDA_FUNCTION_REDIRECTS="DockerDocsRedirectFunction-prod"
```

Reply:
@VictorBersy Thanks, I updated the config with prod config. Is `https://docs2.docker.com` the right domain?

Reply:
Yes it is, we use `docs2.docker.com` for now.